vpr/src/device/rr_graph_builder.h

```diff
@@ -44,6 +44,27 @@ class RRGraphBuilder {
     inline void set_node_type(RRNodeId id, t_rr_type type) {
         node_storage_.set_node_type(id, type);
     }
+    /* Add a edge to the RRGraph, by providing the source and sink node
+     * This function will automatically create a node and
+     * configure the nodes and edges in connection
+     */
+    RRSegmentId add_rr_segment(const t_segment_inf& segment_info);
+    RRSwitchId add_rr_switch_inf(const t_rr_switch_inf& switch_info);
+
+    /* Reserve the lists of nodes, edges, switches etc. to be memory efficient.
+     * This function is mainly used to reserve memory space inside RRGraph,
+     * when adding a large number of nodes/edge/switches/segments,
+     * in order to avoid memory fragements
+     */
+    void reserve_segments(const size_t& num_segments);
+    inline void reserve_edges(size_t num_edges) {
+        node_storage_.reserve_edges(num_edges);
+    }
+
+    size_t num_rr_segments() const;
+    size_t num_rr_switch() const;
```

Comment:
Follow the uniform naming convention here:
- ``num_rr_switch()`` -> ``num_rr_switches()``